src/Features/Core/Portable/Completion/CompletionTriggerKind.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Completion
+{
+    /// <summary>
+    /// The kind of action that triggered completion to start.
+    /// </summary>
+    public enum CompletionTriggerKind
+    {
+        /// <summary>
+        /// Completion was triggered via some other mechanism.
+        /// </summary>
+        Other = 0,
```

Comment:
I think 'other' is too vague.  Knowing if it was triggered by a completion command or not would be valuable
as would knowing if was triggered by teh snippets command.